tuna.go

```diff
@@ -560,3 +563,92 @@ func LoadOrCreateAccount(walletFile, passwordFile string) (*vault.Account, error
 	}
 	return wallet.GetDefaultAccount()
 }
+
+func sendNanoPayment(np *nkn.NanoPay, session *smux.Session, w *nkn.Wallet, cost *common.Fixed64, c *Common, nanoPayFee string) (int, error) {
+	var err error
+	paymentReceiver := c.GetPaymentReceiver()
+	if np == nil || np.Address() != paymentReceiver {
+		np, err = w.NewNanoPay(paymentReceiver, nanoPayFee, DefaultNanoPayDuration)
+		if err != nil {
+			return 0, err
+		}
+	}
+	tx, err := np.IncrementAmount(cost.String())
+	if err != nil {
+		return 0, err
+	}
+	txData := tx.ToArray()
+	stream, err := session.OpenStream()
+	if err != nil {
+		return 0, err
+	}
+	n, err := stream.Write(txData)
+	stream.Close()
+	if !(n == len(txData) && err == nil) {
```

Comment:
If `n != len(txData)`, it's better to return an error explicitly, so `sendNanoPayment` only needs to return an error